toc/rfc/rfc-draft-pcap-bosh.md

```diff
@@ -2,60 +2,214 @@
+* Comprehensive logging of requests, connection issues, congestion and successful completion.
+* mTLS encrypted communication between all components, using gRPC over HTTP/2
+* Authentication and authorization are handled via the existing BOSH UAA and respective BOSH CLI tokens. Accordingly, the current solution is limited to using UAA and token based authentication with RSA tokens. Other authentication mechanisms can be implemented but require additional work.
+
+While the pcap-release is complex, it is also very flexible and versatile, with many advanced use cases that go beyond simply capturing network traffic to a client machine. While these benefits are not realized with the current state of the implementation, they are on the horizon. 
+
+The largest short-term benefit of integrating pcap-release with BOSH is the increased user base and feedback channel that will help improve pcap-release for the mid- and long-term use and future developments.
+
+### Add "pcap-lite", a thin front-end and merge functionality of tcpdump streams
 
-### Integrate PCAP CLI as a "pcap" keyword into BOSH CLI
+The alternative solution is based on the orchestration of existing tools with a small amount of new code.
 
-- This would allow PCAP BOSH CLI to share context with BOSH CLI and have access to the user's login session as well as the currently used BOSH environment.
+Akin to the BOSH SSH feature, a BOSH pcap feature is proposed with similar but slightly different workflow. The well-known `tcpdump` tool is launched on each VM via SSH and transmits its data via the SSH channel. In order to handle multiple VMs and merge their captures, multiple SSH sessions to the respective targets are opened in parallel, multiplexed in the bosh-cli and available for writing to disk or an output stream.
 
-- Another benefit would be that users would not have to download the PCAP BOSH CLI separately. 
+#### Architecture Overview
+
+The architecture of the pcap-lite solution looks as follows:
+
+![](rfc-draft-pcap-bosh/tcpdump-bosh-pcap-lite.drawio.svg "Architecture Overview for the pcap-lite Solution")
+
+**Figure 2 - Architecture Overview for the pcap-lite Solution**
+
+The main benefit of operator convenience over raw use of command line tools requires the following:
+* Automatic determination of VM target addresses via the BOSH Director.
+* A single SSH session can contain multiple streams, one for data, one for process control. The Go SSH library always opens such a "Master Mode" connection.
+* Multiplexing can be done in the BOSH CLI, using a similar mechanism to pcap-release, i.e. by using the `gopacket/gopacket` libraries (and `libpcap` under the hood).
+
+The largest constraint is that the capturing is done used by tools that must already be available on the target VMs, e.g. `tcpdump` that is available as part of the stemcell.
```

Comment:
we should mention input sanitation here, as we are no longer calling a well-defined API but forwarding user input directly to a shell command. as root.

Reply:
But the whole point is that it doesn't make sense to provide those safe-guards. In this scenario you are admin anyway, why bother crafting some sort of injection attack when you can just `bosh ssh -c 'sudo rm -rf /` right away?